File: zig/src/openvpn/internal/logging.zig
// SPDX-FileCopyrightText: 2026 Davide De Rosa
//
// SPDX-License-Identifier: GPL-3.0

const core_mod = @import("../../core/exports.zig");

const log = core_mod.logging;

pub fn sensitiveString(
    level: log.Level,
    comptime prefix: []const u8,
    value: []const u8,
) void {
    log.writef(level, prefix ++ "{s}", .{
        if (log.logsPrivateData()) value else "<redacted>",
    });
}

File: cross/android/io/partout/Logging.kt
// SPDX-FileCopyrightText: 2026 Davide De Rosa
//
// SPDX-License-Identifier: GPL-3.0

package io.partout

import android.util.Log

internal const val REDACTED_VALUE = "<redacted>"

internal fun Any?.sensitiveDescription(logsPrivateData: Boolean): String =
    if (logsPrivateData) toString() else REDACTED_VALUE

internal fun logSensitiveError(
    logTag: String,
    message: String,
    error: Throwable,
    logsPrivateData: Boolean
) {
    if (logsPrivateData) {
        Log.e(logTag, message, error)
    } else {
        Log.e(logTag, "$message (${error.javaClass.name})")
    }
}
